core/src/main/scala/com.snowplowanalytics.elasticsearch/loader/NsqSourceExecutor.scala

```diff
@@ -0,0 +1,124 @@
+import com.github.brainlag.nsq.lookup.DefaultNSQLookup
+import com.github.brainlag.nsq.NSQMessage
+import com.github.brainlag.nsq.NSQConfig
+import com.github.brainlag.nsq.callbacks.NSQMessageCallback
+import com.github.brainlag.nsq.callbacks.NSQErrorCallback
+import com.github.brainlag.nsq.exceptions.NSQException
+
+//Java
+import java.nio.charset.StandardCharsets.UTF_8
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Logging
+import org.slf4j.LoggerFactory
+
+// This project
+import sinks._
+import clients._
+import StreamType._
+
+/**
+ *
+ * @param streamType the type of stream, good/bad
+ * @param documentIndex the elasticsearch index name
+ * @param documentType the elasticsearch index type
+ * @param nsqConfig the NSQ configuration
+ * @param goodSink the configured GoodSink
+ * @param badSink the configured BadSink
+ * @param elasticsearchSender function for sending to elasticsearch
+ */
+class NsqSourceExecutor(
+  streamType: StreamType,
+  documentIndex: String,
+  documentType: String,
+  nsqConfig: ElasticsearchSinkNsqConfig,
+  goodSink: Option[ISink],
+  badSink: ISink,
+  elasticsearchSender: ElasticsearchSender
+) extends Runnable {
+
+  lazy val log = LoggerFactory.getLogger(getClass())
+
+  // Use same transformer with the stdin
+  // because records are coming in same format basically
+  val transformer = streamType match {
+    case StreamType.Good => new SnowplowElasticsearchTransformer(documentIndex, documentType)
+    case StreamType.Bad => new BadEventTransformer(documentIndex, documentType)
+    case StreamType.PlainJson => new PlainJsonTransformer(documentIndex, documentType)
+  }
+
+  val topicName = streamType match {
+    case StreamType.Good => nsqConfig.nsqGoodSourceTopicName
+    case StreamType.Bad => nsqConfig.nsqBadSourceTopicName
+    case StreamType.PlainJson => nsqConfig.nsqPlainJsonTopicName
+  }
+
+  val channelName = streamType match {
+    case StreamType.Good => nsqConfig.nsqGoodSourceChannelName
+    case StreamType.Bad => nsqConfig.nsqBadSourceChannelName
+    case StreamType.PlainJson => nsqConfig.nsqPlainJsonTopicName
+  }
+
+ /**
+   * Consumer will be started to wait new message.
+   */
+  override def run(): Unit = {
+    val nsqCallback = new  NSQMessageCallback {
+      override def message(msg: NSQMessage): Unit = {
+        val msgStr = new String(msg.getMessage(), UTF_8)
+        val emitterInput = transformer.consumeLine(msgStr)
+        emitterInput._2.bimap(
+          f => badSink.store(BadRow(emitterInput._1, f).toCompactJson, None, false),          
+          s => goodSink match {
+            case Some(gs) => gs.store(s.getSource, None, true)
+            case None => elasticsearchSender.sendToElasticsearch(List(msgStr -> s.success))
+          }
+        )
+        msg.finished()
+      }
+    }
+
+    val errorCallback = new NSQErrorCallback {
+      override def error (e: NSQException) = {
+        log.error(s"Caught throwable while consuming in the NSQ", e)
```

Comment:
if there is no string interpolation, no need for the `s`.

Let's modify this to: `s"Exception while consuming topic $topicName"`